src/nc/core/image/Image.cpp

```diff
@@ -108,6 +108,13 @@ void Image::setDemangler(std::unique_ptr<mangling::Demangler> demangler) {
     demangler_ = std::move(demangler);
 }
 
+void Image::setEntryPoint(ByteAddr address) {
+    auto section = getSectionContainingAddress(address);
+
+    if (section && section->isExecutable()) {
```

Comment:
Why do you need this check?
It prevents from setting the entry point before the corresponding code section is added.
This restriction seems artificial to me.

Otherwise, good job, thanks!